src/core/components/collapsible/collapsible.test.tsx

```diff
@@ -0,0 +1,146 @@
+        const buttonLabelOpened = 'Open';
+        const buttonLabelClosed = 'Closed';
+
+        render(createTestComponent({ buttonLabelOpened, buttonLabelClosed }));
+
+        const button = screen.getByText('Closed');
+        await userEvent.click(button);
+        expect(button.textContent).toBe('Open');
+        await userEvent.click(button);
+        expect(button.textContent).toBe('Closed');
+    });
+
+    it('renders open when defaultOpen is true', () => {
+        const children = 'Default Children';
+        const defaultOpen = true;
+        const buttonLabelOpened = 'Open';
+        const buttonLabelClosed = 'Closed';
+        render(createTestComponent({ children, buttonLabelOpened, buttonLabelClosed, defaultOpen }));
+        const button = screen.getByRole('button');
+        expect(button.textContent).toBe(buttonLabelOpened);
+        expect(button.textContent).not.toBe(buttonLabelClosed);
+    });
+
+    it('calls the onToggle callback with the new state', async () => {
+        const onToggle = jest.fn();
+        render(createTestComponent({ onToggle }));
+
+        const button = screen.getByRole('button');
+        await userEvent.click(button);
+        expect(onToggle).toHaveBeenCalledWith(true);
+        await userEvent.click(button);
+        expect(onToggle).toHaveBeenCalledWith(false);
+    });
+
+    it('renders custom button labels', async () => {
+        const buttonLabelOpened = 'Collapse';
+        const buttonLabelClosed = 'Expand';
+        render(createTestComponent({ buttonLabelOpened, buttonLabelClosed }));
+
+        expect(screen.getByText('Expand')).toBeInTheDocument();
+        await userEvent.click(screen.getByText('Expand'));
+        expect(screen.getByText('Collapse')).toBeInTheDocument();
+    });
+
+    it('handles absence of buttonVariant using default button styles', async () => {
+        const buttonLabelOpened = 'Collapse';
+        const buttonLabelClosed = 'Expand';
+        render(createTestComponent({ buttonLabelOpened, buttonLabelClosed }));
+
+        const button = screen.getByRole('button');
+
+        await userEvent.click(button);
+        expect(button).toHaveTextContent('Collapse');
+    });
+
+    it('renders buttonVariant correctly', () => {
+        const buttonVariant = 'primary';
+        const buttonLabelClosed = 'Expand';
+        render(createTestComponent({ buttonVariant, buttonLabelClosed }));
+
+        const button = screen.getByRole('button');
+        expect(button).toHaveClass('bg-primary-400');
+        expect(button).toHaveTextContent('Expand');
+    });
+
+    it('shows overlay correctly instead of base footer when enabled', () => {
+        const showOverlay = true;
+        render(createTestComponent({ showOverlay }));
+        const button = screen.getByRole('button');
+        // eslint-disable-next-line testing-library/no-node-access
+        expect(button.parentElement).not.toHaveClass('mt-4');
```

Comment:
This is really borderline because it's testing that a class isn't there and not that an overlay is present.